File: UI/src/services/costService.js
// Cost service — maps GET /api/Costs
// API CostDto fields match UI shape exactly:
//   id (uuid string), name (string|null), amount (double), category (string|null)
import { get } from './api.js';

const normalise = (c) => ({
  ...c,
  name:     c.name     ?? '',
  category: c.category ?? '',
});

export const getAll = () =>
  get('/Costs').then((data) => data.map(normalise));

File: API/PalletTimeLine.Api/Application/Services/CostService.cs
using Microsoft.EntityFrameworkCore;
using PalletTimeLine.Api.Application.DTOs;
using PalletTimeLine.Api.Infrastructure.Data;

namespace PalletTimeLine.Api.Application.Services;

public class CostService : ICostService
{
    private readonly PalletTimelineDbContext _db;

    public CostService(PalletTimelineDbContext db)
    {
        _db = db;
    }

    public async Task<IReadOnlyList<CostDto>> GetAllAsync(CancellationToken ct = default)
    {
        return await _db.Costs.AsNoTracking()
            .Select(c => new CostDto(c.Id, c.Name, c.Amount, c.Category))
            .ToListAsync(ct);
    }
}
